src/evaluate.jl

```diff
@@ -0,0 +1,70 @@
+function evaluate_lda!(func::Functional, ρ::Array{Float64}; kwargs...)
+    @assert func.family == family_lda
+    n_p = Int(length(ρ) / func.n_spin)
+
+    sizes = Dict(:E => n_p, :Vρ => 2n_p, :V2ρ2 => 3n_p, :V3ρ3 => 4n_p)
+    if func.n_spin == 1
+        for key in keys(sizes)
+            sizes[key] = n_p
+        end
+    end
+
+    dargs = Dict(kwargs)
+    for key in keys(dargs)
+        if !haskey(sizes, key)
+            throw(ArgumentError("Unknown keyword argument: $(string(key))"))
+        end
+        if length(dargs[key]) != sizes[key]
+            throw(DimensionMismatch("Length of keyword argument $(string(key)) == " *
+                                    "$(length(dargs[key])) does not agree with expected " *
+                                    "value $(sizes[key])"))
+        end
+    end
+
+    ccall((:xc_lda, libxc), Cvoid, (Ptr{XCFuncType}, Cint, Ptr{Float64}, Ptr{Float64},
+                                    Ptr{Float64}, Ptr{Float64}, Ptr{Float64}),
+          func.pointer, n_p, ρ, get(dargs, :E, C_NULL),
+          get(dargs, :Vρ, C_NULL), get(dargs, :V2ρ2, C_NULL), get(dargs, :V3ρ3, C_NULL)
+    )
+end
+
+
+function evaluate_gga!(func::Functional, ρ::Array{Float64}, σ::Array{Float64}; kwargs...)
+    @assert func.family == family_gga
+    n_p = Int(length(ρ) / func.n_spin)
+
+    n_σ = 3n_p
+    sizes = Dict(:E => n_p, :Vρ => 2n_p, :Vσ => 3n_p,
```

Comment:
I'm not familiar with calling xc functions when dealing with spin polarized system. 
So `sizes` dict here is used for `spin=2` situation?

Reply:
That's the idea. I have not used libxc with `spin=2` either, but I followed the documentation to check for the right size.